File: 2022/2022W02/PreppindataW02.py
import pandas as pd

final = pd.read_csv(r'C:/Data/PreppinData/PD 2022 Wk 1 Input - Input.csv',
                    parse_dates=['Date of Birth'], date_parser=lambda x: pd.to_datetime(x, format='%m/%d/%Y'))
final = final[['pupil first name', 'pupil last name', 'Date of Birth']]
final["Pupil's Name"] = final['pupil first name']+' '+final['pupil last name']
final["This Year's Birthday"] = final['Date of Birth'].apply(lambda x: x.replace(year = 2022))
final['Month'] = final["This Year's Birthday"].dt.strftime('%B')
final['Cake Needed On'] = final["This Year's Birthday"].dt.strftime('%A')
final = final.replace({'Cake Needed On':{'Saturday':'Friday','Sunday':'Friday'}})
final['BDs per Weekday and Month'] = final["Pupil's Name"].groupby([final['Month'], final['Cake Needed On']]).transform('count')
final['Date of Birth'] = final['Date of Birth'].dt.date
final["This Year's Birthday"] = final["This Year's Birthday"].dt.date
final = final[["Pupil's Name", "Date of Birth", "This Year's Birthday", "Month", "Cake Needed On", "BDs per Weekday and Month"]]

File: 2022/2022W02/preppindataW02.R
library(dplyr)
library(readr)

Sys.setlocale("LC_ALL","English")

final <- read_csv("C:/Data/PreppinData/PD 2022 Wk 1 Input - Input.csv", 
                  col_types = cols(`Date of Birth` = col_date(format = "%m/%d/%Y"))) %>%
  select(c('pupil last name', 'pupil first name', 'Date of Birth')) %>%
  mutate("Pupil\'s Name" = paste(`pupil first name`, `pupil last name`),
         "This Year's Birthday" = `year<-`(`Date of Birth`, 2022),
         "Month" = strftime(`This Year's Birthday`, '%B'),
         "Weekday" = strftime(`This Year's Birthday`, '%A'),
         "Cake Needed On" =  replace(`Weekday`, `Weekday` %in% c("Saturday","Sunday"), "Friday")) %>%
  group_by(`Month`, `Cake Needed On`) %>%
  mutate("BDs per Weekday and Month" = n()) %>%
  select(c("Pupil's Name", "Date of Birth", "This Year's Birthday", "Month", "Cake Needed On", "BDs per Weekday and Month"))
  
View(final)
